integration/auth/auth_test.go

```diff
@@ -53,12 +61,379 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		log.Fatalln(err)
 	}
-
 	os.Exit(m.Run())
 }
 
+func TestUserManagement(t *testing.T) {
+	t.Run("clean the slate", cleanupUsers)
+	t.Run("add some users", populateSomeUsers)
+	t.Run("get user", testGetUser)
+	t.Run("user iterator test", testUserIterator)
+	t.Run("paging iterator test", testIterPage)
+	t.Run("disable", testDisableUser)
+
+	t.Run("update user", testUpdateUser)
+	t.Run("disable", testDisableUser)
+
+	t.Run("remove Display Name", testRemoveDisplayName)
+	t.Run("remove PhotoURL", testRemovePhotoURL)
+	t.Run("remove PhoneNumber", testRemovePhone)
+
+	t.Run("Remove custom claims", testRemoveCustomClaims)
+	t.Run("add custom claims", testAddCustomClaims)
+
+	t.Run("deleting a user", testIterPage)
+	t.Run("delete all users", cleanupUsers)
+}
+
+func populateSomeUsers(t *testing.T) {
+	for i := 0; i < 3; i++ {
+		u, err := client.CreateUser(context.Background(), &auth.UserParams{UID: ptr.String(fmt.Sprintf("user--%d", i))})
+		if err != nil {
+			t.Error("trouble creating", i, err)
+		}
+		testFixtures.uidList = append(testFixtures.uidList, u.UID)
+	}
+	u, err := client.CreateUser(context.Background(), &auth.UserParams{})
+	if err != nil {
+		t.Error(err)
+	}
+	testFixtures.sampleUserBlank = u
+
+	uid := "tefwfd1234"
+	u, err = client.CreateUser(context.Background(), &auth.UserParams{
+		UID:          ptr.String(uid),
+		Email:        ptr.String(uid + "eml5f@test.com"),
+		DisplayName:  ptr.String("display_name"),
+		Password:     ptr.String("assawd"),
+		CustomClaims: map[string]interface{}{"asssssdf": true, "asssssdfdf": "ffd"},
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+	testFixtures.sampleUserWithData = u
+}
+
+func cleanupUsers(t *testing.T) {
+	iter := client.Users(context.Background(), "")
+loop:
+	for {
+		user, err := iter.Next()
+		switch err {
+		case nil:
+			//uids = append(uids, user.UID)
+			err := client.DeleteUser(context.Background(), user.UID)
+			if err != nil {
+				t.Errorf("error deleting uid %s, %s", user.UID, err)
+			}
+		case iterator.Done:
+			break loop
+		default:
+			t.Error(err)
+		}
+	}
+
+}
+
+func testUserIterator(t *testing.T) {
+	iter := client.Users(context.Background(), "")
+	var uids []string
+	gotCount := 0
+	for {
+		u, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		gotCount++
+		uids = append(uids, u.UID)
+	}
+	if gotCount != 5 {
+		t.Errorf("expecting 5 users got %d", gotCount)
```

Comment:
Please follow the error reporting conventions used in other tests. Here and else where.